diff --git a/CMakeLists.txt b/CMakeLists.txt
index 307706a..5e71424 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,9 @@ if(POLICY CMP0167)
 endif()
 
 find_package(Boost COMPONENTS program_options REQUIRED)
+set(fmt_DIR "/usr/lib/${CMAKE_LIBRARY_ARCHITECTURE}/cmake/fmt" CACHE PATH "Ubuntu system fmt" FORCE)
+find_package(fmt CONFIG REQUIRED)
+find_package(spdlog CONFIG REQUIRED)
 
 include_directories(
   /usr/local/include/ddscxx
@@ -25,7 +27,7 @@ include_directories(
   include
 )
 
-link_libraries(unitree_sdk2 ddsc ddscxx rt pthread)
+link_libraries(unitree_sdk2 ddsc ddscxx rt pthread spdlog::spdlog fmt::fmt)
 link_libraries(libboost_program_options.a)
 
 add_executable(inspire_g1 inspire_g1.cpp)
